problem to push images in registry v2

Hi,
I'm trying to migrate my images from a v1 registry to a v2 registry. To do so, I use a reverse proxy to add SSL but I'm not using authentication.
I've got no problem to pull images from registry v1 but when I push I've got this error : 
```
[OK] Verified v2 registry (registry.si3si.com) is available
[!!] Please login to registry.si3si.com:
Username:
FATA[0001] Error response from daemon: no successful auth challenge for https://registry.si3si.com/v2/ - errors: []
``` 
I also tried to add NO_LOGIN=true env variable but in this case I've got another error : 
```
cat: /root/.dockercfg: No such file or directory

[ERROR] Non-zero exit code
[ERROR] Migration from v1 to v2 failed!
```

I've got no other idea of what I can tried

If you need any other information, do not hesitate

Gilles DEHAUDT

--- a/migrator.sh
+++ b/migrator.sh
@@ -477,8 +477,8 @@ main() {
   # check to see if NO_LOGIN is set
   if [ "${NO_LOGIN}" != "true" ]; then
     docker_login ${V1_REGISTRY} ${V1_USERNAME} ${V1_PASSWORD} ${V1_EMAIL}
+    decode_auth ${V1_REGISTRY}
   fi
-  decode_auth ${V1_REGISTRY}
   query_source_images
   show_source_image_list
   pull_images_from_source

